### PR TITLE
JAMES-2045 Add test on date head without century

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java [hunk 1]
```diff
@@ -361,6 +361,32 @@ public void dateFromHeaderShouldBeUsedIfPresent() throws Exception {
     }
 
     @Test
+    public void dateFromHeaderShouldUseCurrentCenturyWhenNone() throws Exception {
```
Comment (strokyl) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java]: Could you add a test with 99 if we decide that 99 should go to 1999 and not 2099

Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessageFactoryTest.java]: same question, same answer

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
This still fails because the Lenient parser is used.

```
        FieldParser<DateTimeField> dateTimeParser = DateTimeFieldLenientImpl.PARSER;
        setFieldParser(FieldName.DATE, dateTimeParser);
        setFieldParser(FieldName.RESENT_DATE, dateTimeParser);
```

Sorry @aduprat , you did some JJ for nothing ;-)

**Comment by chibenwa** [comment]
https://github.com/linagora/james-mime4j/pull/9 solves these issues.

@aduprat  did modifications only on one implementation of DateField.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Just merged

**Pull request closed without merging**